Simplenote/src/main/java/com/automattic/simplenote/PreferencesFragment.java

```diff
@@ -456,6 +484,109 @@ public int compare(String text1, String text2) {
+
+        try {
+            String noteContent = readFile(uri);
+            if ("json".equals(getFileExtension(uri))) {
+                JSONObject jsonData = new JSONObject(noteContent);
+                JSONArray activeNotes = jsonData.getJSONArray("activeNotes");
+                JSONArray trashedNotes = jsonData.getJSONArray("trashedNotes");
+                List<String> trashedIds = new ArrayList<>();
+                JSONArray notesArray = new JSONArray();
+                for (int i = 0; i < activeNotes.length(); i++) {
+                    notesArray.put(activeNotes.getJSONObject(i));
+                }
+                for (int j = 0; j < trashedNotes.length(); j++) {
+                    notesArray.put(trashedNotes.getJSONObject(j));
+                    trashedIds.add(trashedNotes.getJSONObject(j).getString("id"));
+                }
+                for (int k = 0; k < notesArray.length(); k++) {
+                    JSONObject noteJson = notesArray.getJSONObject(k);
+                    Note note = noteBucket.newObject();
+                    Iterator<String> keys = noteJson.keys();
+                    while (keys.hasNext()) {
+                        String key = keys.next();
+                        switch (key) {
+                            case "id":
+                                note.setDeleted(trashedIds.contains(noteJson.getString(key)));
+                                break;
+                            case "content":
+                                note.setContent(noteJson.getString(key));
+                                break;
+                            case "creationDate":
+                                note.setCreationDate(DateTimeUtils.getDateCalendar(noteJson.getString(key)));
+                                break;
+                            case "lastModified":
+                                note.setModificationDate(DateTimeUtils.getDateCalendar(noteJson.getString(key)));
+                                break;
+                            case "tags":
+                                note.setTags(jsonToList(noteJson.getJSONArray(key)));
+                                break;
+                            case "pinned":
+                                note.setPinned(noteJson.getBoolean(key));
+                                break;
+                            case "markdown":
+                                note.setMarkdownEnabled(noteJson.getBoolean(key));
+                                break;
+                            case "publicURL":
+                                // note.setPublished(?);
+                                break;
+                        }
+                    }
+                    note.save();
+                }
+            } else {
+                Note note = noteBucket.newObject();
+                note.setContent(noteContent);
+                note.setCreationDate(Calendar.getInstance());
+                note.setModificationDate(note.getCreationDate());
```

Comment:
I'll have to check, but I believe that we'll want to fill in default values, such as an empty list of tags and system tags

Reply:
Ok, I thought that these fields are already filled by default when creating the note object, so I figured it might look like extra code. Should I add filling tags?

Reply:
if they already get defaults it should be fine. looks like you beat me to confirming that 😉

Reply:
😁 I'll check again just in case